Tests/L0/Maven/_suite.ts

```diff
@@ -1791,15 +1776,9 @@ describe('Maven Suite', function () {
         process.exit = function() { processExitInvoked++; return; };
 
         // Act
-        return VstsServerUtils.breakBuildIfQualityGateFails(mockRunSettings, analysisMetrics)
-            .then(() => {
-                // unhook captured functions since they are important system functions
-                capturedStream.unhook();
-                process.exit = capturedExit;
-
-                // Assert
-                assert(processExitInvoked == 0, `Expected process to have not exited. Actual: ${processExitInvoked}`);
-                return true;
+        return analysisMetrics.getTaskResultFromQualityGateStatus()
+            .then((taskResult) => {
```

Comment:
While it would be far better to use the actual TaskResult enum, referencing it triggers the lazy evaluation of further test code which fails unless the environment variable MOCK_RESPONSES is set (which both cannot and should not be done in this scenario).